lib/Controller/Api1Controller.php

```diff
@@ -728,19 +728,14 @@ public function updateShareDisplayMode(int $shareId, int $displayMode, string $t
 	#[OpenAPI(scope: OpenAPI::SCOPE_DEFAULT)]
 	public function indexTableColumns(int $tableId, ?int $viewId): DataResponse {
 		try {
-			try {
-				// permission check in service class
-				// they might be missing legally when only the view was shared
-				$columns = $this->columnService->findAllByTable($tableId);
-			} catch (PermissionError $e) {
-				if (!$viewId) {
-					throw $e;
-				}
+			if ($viewId) {
```

Comment:
Changes in this file (or in the cypress ones) are OK refactorings, but should not be mixed into a PR that is about something else. Generally, such changes blow the PR and makes it hard to review, and might be confusing in retrospective by being out of the context. In future, please keep them for a separate PR 🙂

Reply:
This exact is not refactoring. It's needed for case when table owner tries to edit his own view. There was a bug that we've loaded just raw view columns without enriching them with `readonly` property. So, table owner can't get correct value for view and overrides them during saving edit form

cypress change is indeed little bit offtopic